rebel-extension/public/scripts/weekly-reminders.js

```diff
@@ -0,0 +1,42 @@
+// background.js
+export function weeklyReminders(message, sendResponse) {
+  const assignments = message.assignments;
+
+  if (!Array.isArray(assignments)) {
+    console.error("Expected assignments to be an array, got:", assignments);
+    sendResponse({});
+    return;
+  }
+
+  //ai chatGpt 4o start
+  const now = new Date();
+  const weekStart = new Date(now);
+  weekStart.setDate(now.getDate() - ((now.getDay() + 6) % 7)); // Monday
+  const weekEnd = new Date(weekStart);
+  weekEnd.setDate(weekStart.getDate() + 6); // Sunday
+  //chatGpt end
```

Comment:
```suggestion
  // ai-gen end
```
Change the AI generation comments to match project format

Reply:
Done